src/client/dfuse/ops/readdir.c

```diff
@@ -33,15 +33,16 @@ void
 dfuse_cb_readdir(fuse_req_t req, struct dfuse_inode_entry *inode,
 		 size_t size, off_t offset)
 {
-	size_t		b_offset = 0;
-	daos_anchor_t	anchor = {};
-	uint32_t	nr = NUM_DIRENTS;
-	struct dirent	dirents[NUM_DIRENTS];
-	struct stat	stbuf;
-	int		i;
-	void		*buf;
-	int		ns;
-	int		rc;
+	struct dfuse_projection_info	*fs_handle = fuse_req_userdata(req);
+	size_t				b_offset = 0;
+	daos_anchor_t			anchor = {};
+	uint32_t			nr = NUM_DIRENTS;
```

Comment:
While search for the value of NUM_DIRENTS, i found that it is defined twice in the source code. Is that done on purpose? Do we actually support different values for each of this definition? It is really error-prone IMO.

Reply:
Both dfuse and dfuse_hl need work on readdir(), as they currently both only support NUM_DIRENTS files in a directory.  The defines are both in C files, and both functions need to be reworked to avoid using a max number anyway so the name conflict itself doesn't concern me.